packages/host/app/resources/room.ts

```
@@ -356,34 +368,41 @@ export class RoomResource extends Resource<Args> {
     return event.content.msgtype === APP_BOXEL_REALM_SERVER_EVENT_MSGTYPE;
   }
 
-  private async createSkillCard(cardDoc: LooseSingleCardDocument) {
-    let cardId = cardDoc.data.id;
+  private async createSkillCard(doc: LooseSingleCardDocument) {
+    let cardId = doc.data.id;
     if (!cardId) {
-      console.warn(
-        `No card id found, this should not happen, this can happen if you add a skill card to a room without saving it, and without giving it an ID`,
+      throw new Error(
+        `SKill card document has no id, this should not happen: ${JSON.stringify(doc, null, 2)}`,
       );
```

If this shouldn't happen we should throw an error so we can see that it is not happening and take steps to fix that